build/lender/Lender.sol

```diff
@@ -21,15 +21,21 @@ contract Lender {
   event Lend(uint8 principal, address indexed underlying, uint256 indexed maturity, uint256 returned);
   event Mint(uint8 principal, address indexed underlying, uint256 indexed maturity, uint256 amount);
 
-  /// @param i the deployed Illuminate contract
-  constructor(address i, address s, address p, address t) {
+  /// @param s: the swivel contract
+  /// @param p: the pendle contract
+  /// @param t: the tempus contract
+  constructor(address s, address p, address t) {
     admin = msg.sender;
-    illuminate = i; // TODO add an authorized setter for this?
     swivelAddr = s;
     pendleAddr = p;
     tempusAddr = t;
   }
 
+  function setIlluminate(address i) external {
```

Comment:
let's follow the Swivel pattern and use `..Address` in the name

https://github.com/Swivel-Finance/swivel/blob/main/contracts/v2/marketplace/MarketPlace.sol#L39=

so `setIlluminateAddress`

also

follow the swivel pattern and require the address to never have been set. "mutable once" kind of "immutable" ...

also

this needs to be `authorized` -- you can do the auth modifier stuff in another PR of course.